### PR TITLE
observe: add support for first and follow at the same time

### PR DESCRIPTION
**Pull request by sladyn98**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

Support -f and --first at the same time

Fixes: #959

```release-note
Support -f and --first at the same time
<!-- Enter the release note text here if needed or remove this section! -->
```

### CODE REVIEW

**Comment by sladyn98** [comment]
CC @pchaigno

**Comment by sladyn98** [comment]
@kaworu I need some help with the test to be added. If I get it correctly we would need to test if the first and follow are supported at the same time in the test right

File: pkg/hubble/observer/local_observer.go [hunk 1]
```diff
@@ -595,9 +595,6 @@ func (r *eventsReader) Next(ctx context.Context) (*v1.Event, error) {
 }
 
 func validateRequest(req genericRequest) error {
-	if req.GetFirst() && req.GetFollow() {
-		return status.Errorf(codes.InvalidArgument, "first cannot be specified with follow")
-	}
 	return nil
 }
```
Comment (kaworu) [inline pkg/hubble/observer/local_observer.go:599]: At that point I think we should remove this `func` as it doesn't do anything useful anymore.

**Review by kaworu — CHANGES_REQUESTED** [review]
> If I get it correctly we would need to test if the first and follow are supported at the same time in the test right

@sladyn98 correct, [the already existing `TestLocalObserverServer_GetFlows_Follow_Since` func](https://github.com/cilium/cilium/blob/1c53367bb9abc1ddffbbc7f01e93d38fa45337cf/pkg/hubble/observer/local_observer_test.go#L228) could be used as inspiration as it test the follow mode.

**Pull request closed without merging**